src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesDeclarativeAgent.java

```diff
@@ -95,20 +113,124 @@ public ContainerTemplate getContainerTemplate() {
         return containerTemplate;
     }
 
+    @DataBoundSetter
+    public void setContainerTemplate(ContainerTemplate containerTemplate) {
+        this.containerTemplate = containerTemplate;
+    }
+
+    public PodTemplate getPodTemplate() {
+        return podTemplate;
+    }
+
+    @DataBoundSetter
+    public void setPodTemplate(PodTemplate podTemplate) {
+        this.podTemplate = podTemplate;
+        List<ContainerTemplate> containers = podTemplate.getContainers();
+        if (containers != null && containers.size() > 0) {
+            this.containerTemplate = containers.get(0);
+        }
+    }
+
     public int getActiveDeadlineSeconds() {
         return activeDeadlineSeconds;
     }
 
     @DataBoundSetter
     public void setActiveDeadlineSeconds(int activeDeadlineSeconds) { this.activeDeadlineSeconds = activeDeadlineSeconds; }
 
+    public String getPodTemplateName() {
+        return podTemplateName;
+    }
+
+    @DataBoundSetter
+    public void setPodTemplateName(String podTemplateName) {
+        this.podTemplateName = podTemplateName;
+        if (podTemplateName != null) {
+            String cloudName = getCloud();
+            if (StringUtils.isEmpty(cloudName)) {
+                cloudName = KubernetesCloud.CLOUD_NAME;
+            }
+            Cloud cloud = Jenkins.getInstance().getCloud(cloudName);
+            if (cloud instanceof KubernetesCloud) {
+                KubernetesCloud kubernetesCloud = (KubernetesCloud) cloud;
+                List<PodTemplate> templates = kubernetesCloud.getTemplates();
+                PodTemplate template = null;
+                for (PodTemplate t : templates) {
+                    String name = t.getName();
+                    if (podTemplateName.equals(name)) {
+                        template = t;
+                        break;
+                    }
+                }
+                if (template != null) {
+                    setPodTemplate(template);
+                }
```

Comment:
it should log something at INFO level if I'm using a name that doesn't match anything, sounds like a typical user error

Reply:
I guess it should be a warning as either the user made a typo or the Jenkins needs to be configured correctly to add the named podTemplate?